### PR TITLE
Add confirmation for URL actions

### PR DESCRIPTION
**Pull request by bramkragten**
## Summary
Adds a confirmation for `call_service`, `fire_event` and `send_location`, `render_template` url action.

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#

### CODE REVIEW

**Comment by cpvbruno** [comment]
Besides missing tests (not sure how's the criteria we use for that) it looks good to me 👍🏻

File: Sources/App/WebView/IncomingURLHandler.swift [hunk 1]
```diff
@@ -150,6 +162,34 @@ class IncomingURLHandler {
         }
     }
 
+    private func confirmAction(title: String, message: String, handler: @escaping () -> Void, cancelHandler: (() -> Void)? = nil) {
```
Comment (zacwest) [inline Sources/App/WebView/IncomingURLHandler.swift]: `-> Promise<Void>` may make the call sites a little easier, e.g.

```swift
confirmAction(…).done {
  // do the thing
}.catch {
  // do an error
}
```

and `.cauterize()` when we don't care about the error, or in the places right now…

```swift
firstly {
  confirmAction(…)
}.done {
  // rest of the chains
}
```

File: Sources/App/WebView/IncomingURLHandler.swift [hunk 2]
```diff
@@ -150,6 +162,34 @@ class IncomingURLHandler {
...
+        let alert = UIAlertController(
+            title: title,
+            message: message,
+            preferredStyle: UIAlertController.Style.alert
+        )
+
+        alert.addAction(UIAlertAction(
+            title: L10n.noLabel,
```
Comment (zacwest) [inline Sources/App/WebView/IncomingURLHandler.swift]: cancel may be a bit more clear

**Review by zacwest — APPROVED** [review]

**Comment by zacwest** [comment]
you'll want to run the linting scripts to make _that_ check happy, at least.

code coverage for some of these areas is missing.

**Review by zacwest — APPROVED** [review]

**Pull request merged**